internal/provider/registry.go

```diff
@@ -0,0 +1,91 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/installcfg"
+	"github.com/openshift/assisted-service/internal/provider/baremetal"
+	"github.com/openshift/assisted-service/internal/provider/vsphere"
+	"github.com/openshift/assisted-service/internal/usage"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+// ErrNoSuchProvider is returned or thrown in panic when the specified provider is not registered.
+var ErrNoSuchProvider = errors.New("no provider with the specified name registered")
+
+// ErrProviderUnSet is returned or thrown in panic when provider has not been set.
+var ErrProviderUnSet = errors.New("provider has not been set")
+
+type ProviderRegistry interface {
+	Registry
+	Provider
+}
+
+// Registry registers the providers to their names.
+type Registry interface {
+	// Register registers a provider.
+	Register(provider Provider)
+	// Get returns a provider registered to a name.
+	// if provider is not registered returns an ErrNoSuchProvider
+	Get(name string) (Provider, error)
+	// GetCurrentProvider returns the current provider.
+	// if provider has not been set returns ErrProviderUnSet.
+	GetCurrentProvider() (Provider, error)
+	// SetCurrentProvider sets the current provider to the specific provider.
+	// if provider is not registered returns an ErrNoSuchProvider
+	SetCurrentProvider(name string) error
```

Comment:
Assisted-service running in parallel with multiple clusters. Each cluster could have a different provider. 
I suggest to remove the `CurrentProvider` functionality although it makes it more simple, we can't have this caching.

Reply:
done